docs/rfcs/XeGPU.md

```diff
@@ -701,6 +700,332 @@ An example on how to perform transpose using load with chunk_size in SIMT flavor
+
+**Attribute xegpu.wg_map**
+
+`wg_map` specifies how a n-d tensor (defined by the tensor descriptor) is partitioned among subgroup within a workgroup. wg_map consists of three parameters:
+  * sg_layout: Defines the n-d arrangement of subgroups within the workgroup. The dimensions can be a 3d array as [dim_0, dim_1, dim_2]. 
+  * sg_data: Specifies the shape of the tensor size for each subgroup after decomposition.
+  * sg_order: The dimension order used to linearize n-d subgroup ids to 1-d. The first dimension in the sg_order list is the fastest-changing dimension. 
+
+Example of linerized subgourp id regarding order[1, 0] vs. order [0, 1]. 
+| sg_layout[4, 4] | order[1, 0] | order[0, 1]
+| :----   | :----   |  :----   |
+| [0, 0], [0, 1], [0, 2], [0, 3] | 0, 1, 2, 3 | 0, 4, 8, 12 |
+| [1, 0], [1, 1], [1, 2], [1, 3] | 4, 5 , 6, 7| 1, 5, 9, 13 |
+| [2, 0], [2, 1], [2, 2], [2, 3] | 8, 9, 10 , 11 | 2, 6, 10, 14 |
+| [3, 0], [3, 1], [3, 2], [3, 3] | 12, 13, 14, 15 | 3, 7, 11, 15 |
+
+For a subgroup threads in 3-d sg_layout [dim_0, dim_1, dim_2], sg_order[2, 1, 0] maps a subgroup thread with 3-d index [x, y, z] to a linear subgroup thread index [z + dim_2*y + dim_2*dim_1*x ], sg_order[1, 2, 0] maps to [y + dim_2*z + dim_2*dim_1*x].
+
+When a wg_map attribute is attached to a tensor descriptor, load/store/dpas will operate at the workgroup level. The wg_map attribute must be specified when creating the tensor descriptor.
+
+**Constraints**
+
+Given these definitions:
+```mlir
+sg_data_size = sg_data[0] × sg_data[1]
+workgroup_size = sg_layout[0] × sg_layout[1]
+tensor_size = tensor_desc[0] × tensor_desc[1]
+```
+
+The following conditions must hold:
+
+* workgroup_size must represent the number of subgroups in a workgroup for a kernel.
+* tensor_desc[0] must be either evenly divisible by sg_layout[0] × sg_data[0], or vice versa.
+* tensor_desc[1] must be either evenly divisible by sg_layout[1] × sg_data[1], or vice versa.
+
+**distribution rule**
+
+The tensor_desc is distributed to sg_data x sg_layout along each dimension in a round-robin fashion. If sg_data[i] x sg_layout[i] < tensor_desc[i], we have data left after all subgroups are assigned for the first round, we continue to assign the rest data starting from the first subgroup until the data is completely assigned. If sg_data[i] x sg_layout[i] > tensor_desc[i], we may have already used up all the data before all subgroups are assigned. In this case, we wrap around the tensor data and continue the assignment, and the rest subgroups along that dimension share the same data.
```

Comment:
Suggest using `wrap-around` to describes sg/wi assigned multiple non-contiguous tiles. And `broadcast` when sg/wi share tiles.

Reply:
Thanks.